### PR TITLE
CVE-2011-1435 and CVE-2016-1623

### PR DESCRIPTION
**Pull request by kdm5498**
Modified the .yml files for CVE's 2011-1435 and 2016-1623.

### CODE REVIEW

File: cves/CVE-2011-1435.yml [hunk 1]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  The implementation of permissions for browser extensions was not properly secured.
+  This allowed an attacker to access the files on the same computer as the browser
+  through the use of a browser extension without having the proper permissions.
```
Comment (siberiax) [inline cves/CVE-2011-1435.yml]: What specifically about the permissions was not correct?

File: cves/CVE-2011-1435.yml [hunk 2]
```diff
@@ -58,7 +61,7 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 99ae5d5069ca7f3a6b681d2b04b719699562ca50
-  :note: ''
+  :note: 'Fixes wording on API descriptions'
 - :commit: 1528b69594e86dcaae51da203a13ac9e61a045f2
   :note: ''
```
Comment (siberiax) [inline cves/CVE-2011-1435.yml]: What does this fix commit do?

File: cves/CVE-2011-1435.yml [hunk 3]
```diff
@@ -58,7 +61,7 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 99ae5d5069ca7f3a6b681d2b04b719699562ca50
-  :note: ''
+  :note: 'Fixes wording on API descriptions'
```
Comment (siberiax) [inline cves/CVE-2011-1435.yml]: This comment should be expanded upon

**Comment by siberiax** [comment]
Good work on 2016-1623!

File: cves/CVE-2011-1435.yml [hunk 4]
```diff
@@ -205,4 +217,7 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    I feel that there was a design mistake made to lead to this vulnerability.
+    The design decision to allow for unrestricted access to the file system with 
+    a lower level permission led to this vulnerability.
```
Comment (Fikayoz) [inline cves/CVE-2011-1435.yml]: Is there any information on WHY this design decision was made? Was it simply oversight, or did they have a purpose in giving such unrestricted access?

**Comment by Fikayoz** [comment]
I didn't find any grammar or research issues for 2016-1623!. Seems pretty solid; good job!
I give this 2016-1623 four upvotes.

File: cves/CVE-2011-1435.yml [hunk 5]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 732
```
Comment (tilley14) [inline cves/CVE-2011-1435.yml]: From your research of the fix, do you think that the browser was designed to intentionally allow file access permissions for authorized extensions? From the issues page and the comment around that was linked to in the thread I thought that CWE-862 may also apply to this CVE.

File: cves/CVE-2011-1435.yml [hunk 6]
```diff
@@ -205,4 +220,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    I feel that there was a design mistake made to lead to this vulnerability.
+    The design decision to allow for unrestricted access to the file system with
```
Comment (tilley14) [inline cves/CVE-2011-1435.yml]: There were some interesting comments from the developers about prioritizing the threat level of this vulnerability. Specifically comments 8, 9, and 10 here https://bugs.chromium.org/p/chromium/issues/detail?id=72523
What are your thoughts?

File: cves/CVE-2011-1435.yml (hunk 5 above)
Comment (kdm5498) [reply cves/CVE-2011-1435.yml]: It seems that the issue was with a permission that specifically allowed file access permissions to authorized extensions, so the browser was intentionally designed with this in mind. I feel that while 862 could also apply, 732 applies more closely, as it was not a case of missing authorization, but more misallocated authorization of resources outside the extension's sandbox.

File: cves/CVE-2016-1623.yml [hunk 7]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 666
```
Comment (tilley14) [inline cves/CVE-2016-1623.yml]: Great find! I Think this CWE is very specific to this situation!

File: cves/CVE-2011-1435.yml (hunk 6 above)
Comment (kdm5498) [reply cves/CVE-2011-1435.yml]: I feel like they were correct in their assessment of the severity of the issue, it was an issue that only affects users who downloaded a malicious extension, and only users that installed an extension with the specific permissions that were affected and so therefore was not a maximum severity threat.

File: cves/CVE-2016-1623.yml [hunk 8]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Due to the implementation of the HTML parser, an unapproved frame component 
+  could attatch itself to the page at the same time an approved component was detatching.
+  
+  This allowed remote attackers to piggyback off the approved frame to execute commands
```
Comment (tilley14) [inline cves/CVE-2016-1623.yml]: I think that slightly rewording this to say that the attackers use, specifically, the **current state** of the approved frame may emphasize the idea that this is an issue caused by operations in the wrong phase of the life cycle.

File: cves/CVE-2016-1623.yml [hunk 9]
```diff
@@ -68,7 +73,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 4
```
Comment (tilley14) [inline cves/CVE-2016-1623.yml]: 4 Ups from me!

File: cves/CVE-2016-1623.yml (hunk 8 above)
Comment (kdm5498) [reply cves/CVE-2016-1623.yml]: Thanks, reworded.

**Pull request merged**